### PR TITLE
Change references from "master" branch to "main" branch

### LINKED ISSUE #1356: Change main branch from "master" to "main"
In the interest of inclusivity, should we change the main git branch from `master` to `main`? This would be in line with changes to large Python repos, including CPython, Numpy, Flask, and Django, as well as the default main branch for all new GitHub repos.

**Are you willing to help implement and maintain this feature?** Yes

### PR DESCRIPTION
**Pull request by willschlitzer**
As discussed in #1356, PyGMT is changing its default branch from "master" to "main". This pull request changes the references to "master" in the CI job files, maintenance.md, and the README.

Fixes #1356

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: .github/workflows/ci_tests_dev.yaml [hunk 1]
```diff
@@ -23,15 +23,15 @@ on:
...
+  test_gmt_main:
     name: ${{ matrix.os }} - GMT ${{ matrix.gmt_git_ref }}
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         python-version: [3.9]
         os: [ubuntu-latest, macOS-11.0, windows-latest]
-        gmt_git_ref: [master]
+        gmt_git_ref: [main]
```
Comment (weiji14) [inline .github/workflows/ci_tests_dev.yaml:34]: This references upstream GMT, so we have to keep it on 'master' for now.

```suggestion
        gmt_git_ref: [master]
```

File: .github/workflows/ci_tests_dev.yaml [hunk 2]
```diff
@@ -90,7 +90,7 @@ jobs:
 
       # Build and install latest GMT from GitHub
       - name: Install GMT ${{ matrix.gmt_git_ref }} branch (Linux/macOS)
-        run: curl https://raw.githubusercontent.com/GenericMappingTools/gmt/master/ci/build-gmt.sh | bash
+        run: curl https://raw.githubusercontent.com/GenericMappingTools/gmt/main/ci/build-gmt.sh | bash
```
Comment (weiji14) [inline .github/workflows/ci_tests_dev.yaml:93]:
```suggestion
        run: curl https://raw.githubusercontent.com/GenericMappingTools/gmt/master/ci/build-gmt.sh | bash
```

File: .github/workflows/dvc-diff.yml [hunk 3]
```diff
@@ -44,7 +44,7 @@ jobs:
       run: |
         echo -e "## Summary of changed images\n" > report.md
         echo -e "This is an auto-generated report of images that have changed on the DVC remote\n" >> report.md
-        dvc diff --show-md master HEAD >> report.md
+        dvc diff --show-md main HEAD >> report.md
```
Comment (weiji14) [inline .github/workflows/dvc-diff.yml:47]: This reminds me, we'll need to rename the 'master' branch on https://dagshub.com/GenericMappingTools/pygmt to 'main' too.

File: doc/maintenance.md [hunk 4]
```diff
@@ -57,7 +57,7 @@ This means that all commits will be collapsed into one.
 The main advantages of this are:
 
 * Eliminates experimental commits or commits to undo previous changes.
-* Makes sure every commit on master passes the tests and has a defined purpose.
+* Makes sure every commit on main passes the tests and has a defined purpose.
```
Comment (weiji14) [inline doc/maintenance.md:60]:
```suggestion
* Makes sure every commit on the main branch passes the tests and has a defined purpose.
```

File: doc/maintenance.md [hunk 5]
```diff
@@ -115,13 +115,13 @@ There are 11 configuration files located in `.github/workflows`:
...
    This workflow is run to publish wheels to PyPI and TestPyPI (for testing only).
-   Archives will be pushed to TestPyPI on every commit to the *master* branch
+   Archives will be pushed to TestPyPI on every commit to the *main* branch
    and tagged releases, and to PyPI for tagged releases only.
 
 7. `release-drafter.yml` (Drafts the next release notes)
 
     This workflow is run to update the next releases notes as pull requests are
-    merged into master.
+    merged into main.
```
Comment (weiji14) [inline doc/maintenance.md:124]:
```suggestion
    merged into the main branch.
```

**Comment by meghanrjones** [comment]
While GitHub will redirect links, I think we should update the links in these files to reflect the new `main` name as part of this process:
```
./doc/install.rst
./doc/conf.py
./pygmt/src/x2sys_cross.py
./doc/contributing.md
./doc/external_resources.md
./.github/config.yml
```

File: .github/workflows/ci_tests_dev.yaml [hunk 6]
```diff
@@ -23,7 +23,7 @@ on:
     - cron: '0 0 * * *'
 
 jobs:
-  test_gmt_master:
+  test_gmt_main:
```
Comment (seisman) [inline .github/workflows/ci_tests_dev.yaml:26]: It means "test the GMT master branch", so it should stay with "master", but I think it's OK to change it to "test_gmt_dev:
```suggestion
  test_gmt_dev:
```

**Comment by willschlitzer** [comment]
> While GitHub will redirect links, I think we should update the links in these files to reflect the new `main` name as part of this process:
> 
> ```
> ./doc/install.rst
> ./doc/conf.py
> ./pygmt/src/x2sys_cross.py
> ./doc/contributing.md
> ./doc/external_resources.md
> ./.github/config.yml
> ```

`./doc/install.rst` and `./pygmt/src/x2sys_cross.py`
It looks like the only instance of `master` is in reference to the GMT branch

**Comment by meghanrjones** [comment]
> > While GitHub will redirect links, I think we should update the links in these files to reflect the new `main` name as part of this process:
> > ```
> > ./doc/install.rst
> > ./doc/conf.py
> > ./pygmt/src/x2sys_cross.py
> > ./doc/contributing.md
> > ./doc/external_resources.md
> > ./.github/config.yml
> > ```
> 
> `./doc/install.rst` and `./pygmt/src/x2sys_cross.py`
> It looks like the only instance of `master` is in reference to the GMT branch

sounds good, I did a quick search but didn't spend much time checking the references.

**Comment by seisman** [comment]
Maybe we should give this PR a final review and merge it several days before the v0.4.1 release, so that we can have more time to check if there are any related issues in the upcoming v0.4.1 release.

File: doc/contributing.md [hunk 7]
```diff
@@ -188,6 +188,7 @@ Before building and testing the project, you have to activate the environment
 conda activate pygmt
 ```
 
+
 We have a [`Makefile`](https://github.com/GenericMappingTools/pygmt/blob/master/Makefile)
```
Comment (core-man) [inline doc/contributing.md:192]:
```suggestion
We have a [`Makefile`](https://github.com/GenericMappingTools/pygmt/blob/main/Makefile)
```

**Comment by core-man** [comment]
`contributing.md` still have sentences using **master**:
- Line 98:
  ```
  our tests. This way, the *master* branch is always stable.
  ```
- Line 173:
  ```
  See the [`environment.yml`](https://github.com/GenericMappingTools/pygmt/blob/master/environment.yml)
  ```

**Review by core-man — APPROVED** [review]
I think all "master" are changed to "main" except in some places indicated above. Great~

**Comment by weiji14** [comment]
Besides those mentioned at https://github.com/GenericMappingTools/pygmt/pull/1360#issuecomment-890307400, here's a few more:

- https://github.com/GenericMappingTools/pygmt/blob/6a4b8715bf00cd42cae8d5405e35e08a4764e97c/doc/conf.py#L117
- https://github.com/GenericMappingTools/pygmt/blob/6a4b8715bf00cd42cae8d5405e35e08a4764e97c/doc/_templates/layout.html#L35
- https://github.com/GenericMappingTools/pygmt/blame/v0.4.0/doc/install.rst#L72

**Comment by willschlitzer** [comment]
@weiji14 @core-man I made the changes you requested

**Comment by seisman** [comment]
> Besides those mentioned at [#1360 (comment)](https://github.com/GenericMappingTools/pygmt/pull/1360#issuecomment-890307400), here's a few more:
> 
> * https://github.com/GenericMappingTools/pygmt/blob/6a4b8715bf00cd42cae8d5405e35e08a4764e97c/doc/conf.py#L117
> * https://github.com/GenericMappingTools/pygmt/blob/6a4b8715bf00cd42cae8d5405e35e08a4764e97c/doc/_templates/layout.html#L35

I don't think we can change `master_doc` to `main_doc`. It's a sphinx configuration and is used internally by sphinx (see https://github.com/sphinx-doc/sphinx/search?q=master_doc).

File: doc/conf.py [hunk 8]
```diff
@@ -114,7 +114,7 @@
 needs_sphinx = "1.8"
 # The encoding of source files.
 source_encoding = "utf-8-sig"
-master_doc = "index"
+main_doc = "index"
```
Comment (weiji14) [inline doc/conf.py:117]:
```suggestion
root_doc = "index"
```

File: doc/_templates/layout.html [hunk 9]
```diff
@@ -32,7 +32,7 @@
 
 
 {% block sidebartitle %}
-    <a href="{{ pathto(master_doc) }}"><h2 class="sidebar-title">{{ project }}</h2></a>
+    <a href="{{ pathto(main_doc) }}"><h2 class="sidebar-title">{{ project }}</h2></a>
```
Comment (weiji14) [inline doc/_templates/layout.html:35]:
```suggestion
    <a href="{{ pathto(root_doc) }}"><h2 class="sidebar-title">{{ project }}</h2></a>
```

**Review by weiji14 — COMMENTED** [review]
> > Besides those mentioned at [#1360 (comment)](https://github.com/GenericMappingTools/pygmt/pull/1360#issuecomment-890307400), here's a few more:
> > 
> > * https://github.com/GenericMappingTools/pygmt/blob/6a4b8715bf00cd42cae8d5405e35e08a4764e97c/doc/conf.py#L117
> > * https://github.com/GenericMappingTools/pygmt/blob/6a4b8715bf00cd42cae8d5405e35e08a4764e97c/doc/_templates/layout.html#L35
> 
> I don't think we can change `master_doc` to `main_doc`. It's a sphinx configuration and is used internally by sphinx (see https://github.com/sphinx-doc/sphinx/search?q=master_doc).
> 
> **EDIT:** Sphinx has renamed `master_doc` to `root_doc` in sphinx v4.0. We can change `master_doc` to `root_doc` here, but it means PyGMT will require sphinx >= 4.0, which was release in May, 2021.

Was just about to say that `root_doc` should be used instead. We have sphinx 4.1.2 in CI (https://github.com/GenericMappingTools/pygmt/runs/3235302620?check_suite_focus=true#step:6:187) so it should be fine.

**Review by weiji14 — APPROVED** [review]

**Review by meghanrjones — APPROVED** [review]
Does this get merged before or after the actual renaming of the branch?

**Review by seisman — APPROVED** [review]

**Comment by seisman** [comment]
> Does this get merged before or after the actual renaming of the branch?

Either should work. I prefer to merge this PR **after** renaming the branch, so that we can immediately know how renaming the main branch affects PRs.

**Comment by meghanrjones** [comment]
> > Does this get merged before or after the actual renaming of the branch?
> 
> Either should work. I prefer to merge this PR **after** renaming the branch, so that we can immediately know how renaming the main branch affects PRs.

OK, I can rename the branch after the current workflows finish and then ping here to merge in the PR.

**Comment by meghanrjones** [comment]
Looks good to merge after checks pass, @willschlitzer!

**Comment by weiji14** [comment]
> Looks good to merge after checks pass, thanks @willschlitzer!

Thanks heaps @willschlitzer! I'll squash and merge this in directly since the "master" -> "main" rename has been done (otherwise the links on https://www.pygmt.org/dev/index.html won't work).

**Pull request merged**